File: backend/src/main/java/com/bv87/diemdanh/dto/StaffPositionDto.java
package com.bv87.diemdanh.dto;

import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class StaffPositionDto {
    private final Integer positionCode;
    private final String positionCodeFormatted;
    private final String positionName;
    private final int sortOrder;
    private final boolean active;
    private final long usageCount;
}

File: desktop/src/BV87.Core/Models/Admin/Catalog/StaffPositionDto.cs
namespace BV87.Core.Models.Admin.Catalog;

public sealed class StaffPositionDto
{
    public int PositionCode { get; set; }
    public string? PositionCodeFormatted { get; set; }
    public string? PositionName { get; set; }
    public int SortOrder { get; set; }
    public bool Active { get; set; }
    public long UsageCount { get; set; }
}

public sealed class StaffPositionUpsertRequest
{
    public string PositionName { get; set; } = string.Empty;
    public int? SortOrder { get; set; }
    public bool Active { get; set; } = true;
}
